mmds/src/data_store.rs

```diff
@@ -94,17 +129,12 @@ impl Mmds {
                     None => {
                         // When the object is not a map, return the value.
                         // The only supported Value type is String.
-                        match val.as_str() {
-                            Some(str_val) => {
-                                ret.push(str_val.to_string());
-                                Ok(ret)
-                            }
-                            None => Err(Error::UnsupportedValueType),
-                        }
+                        ret.push(val.as_str().unwrap().to_string());
+                        Ok(ret)
                     }
                 }
             }
-            None => Err(Error::NotFound),
+            None => Err(()),
```

Comment:
This function's comment says it should return `NotFound` - why has the return type changed?

Reply:
H! I thought since this function won't return a `Err(Error::UnsupportedValueType)` anymore (only string values get accepted), I can remove the wrapped Error Enum in Err().

Reply:
The function can still return `NotFound`. That error should remain in place. It cannot return  `UnsupportedValueType` anymore, but you could still pass a path that is not present in the data store. In this case we should return `NotFound`.